src/components/RecipeList.tsx

```diff
@@ -40,12 +82,63 @@ const RecipeList: React.FC = () => {
     );
   };
 
+  const handlePage = (p: number) => {
+    setPage(p);
+  };
+
+  const recipeFilterOptions = [
+    { value: 3, name: 3 },
+    { value: 6, name: 6 },
+    { value: 10, name: 10 },
+  ];
+
+  const filterHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setNumberElems(Number(e.target.value));
+  };
+
+  const totalPages = Math.ceil(recipeList.recipes.length / numberElems);
+
+  const indexHelper = () => {
+    if (page === 1) {
+      setFirstIndex(0);
+      setLastIndex(numberElems);
+    } else if (page === 2) {
+      setFirstIndex(numberElems);
+      setLastIndex(numberElems * 2);
+    } else {
+      setFirstIndex(page * numberElems - numberElems);
+      setLastIndex(numberElems * page);
+    }
+  };
+
+  useEffect(() => {
+    indexHelper();
+    let array = recipeList.recipes.slice(firstIndex, lastIndex);
+    if (array.length === 0) {
+      setPage(1);
+      setArrayForRender(recipeList.recipes.slice(0, numberElems));
+    } else {
+      setArrayForRender(array);
+    }
+  }, [page, numberElems, lastIndex, firstIndex, recipeList]);
+
+  const reserAddedRecipes = (e: React.MouseEvent) => {
+    e.preventDefault();
+    dispatch(removeRecipes());
+    fetchREcipes();
+  };
+
   const recipesView = () => {
     return (
       <div className="d-flex flex-column align-aitems-center mb-5 mt-5">
         <h1 className="mt-5 text-center">Recipe List</h1>
+        <AmoutnRecipesFilter
+          onChange={filterHandler}
+          options={recipeFilterOptions}
+        />
+        <Btn onClick={(e) => reserAddedRecipes(e)}>Reset added recipes</Btn>
         <ListGroup className="w-100 mx-auto justify-content-center mt-4">
-          {recipes?.recipes.map((recipe) => (
+          {arrayForRender.map((recipe, index) => (
```

Comment:
I guess you were using `index` as a key and then switched to key={recipe.id}
which is great; never use index as a key [https://reactjs.org/docs/lists-and-keys.html#keys]